src/state/RxState.js

```diff
@@ -0,0 +1,63 @@
+import React, { PropTypes, Component } from "react";
+import { Observable } from 'rxjs/Rx';
+
+// Creates a state for use with the connect and provider below
+export function createState(reducer$, initialState$ = Observable.of({})) {
+  return initialState$
+    .merge(reducer$)
+    .scan((state, [scope, reducer]) =>
+      ({ ...state, [scope]: reducer(state[scope]) }))
+    .publishReplay(1)
+    .refCount();
+}
+
+// Used much the same way as redux connect
+export const connect = (selector = state => state) => (providedState$) => {
+  return function wrapWithConnect(WrappedComponent) {
+    return class Connect extends Component {
+      state$ = providedState$;
+
+      static contextTypes = {
+        state$: PropTypes.object // .isRequired
+      };
+
+      constructor(props, context) {
+        super(props, context);
+        this.state$ = this.context.state$ || this.state$;
```

Comment:
Maybe i can swap this OR condition. So that some connected components use their own state